pvactools/lib/valid_algorithms.py

```diff
@@ -0,0 +1,58 @@
+import sys
+import argparse
+
+from pvactools.lib.prediction_class import *
+
+class ValidAlgorithms:
+    def __init__(self, allele, species):
+        self.allele = allele
+        self.species = species
+
+    def print_valid_algorithms(self):
+        if self.allele is None:
+            valid_algorithms = []
+            if self.species is None:
+                valid_algorithms = PredictionClass.prediction_methods()
+            else:
+                prediction_algorithms = PredictionClass.prediction_methods()
+                for algorithm in prediction_algorithms:
+                    cls = globals()[algorithm]
+                    alleles = cls().valid_allele_names()
+                    for allele in alleles:
+                        if cls.species_for_allele(allele) == self.species:
+                            valid_algorithms.append(algorithm)
+                            break
+        else:
+            PredictionClass.check_alleles_valid([self.allele])
+            if (self.species != None and PredictionClass.species_for_allele(self.allele) != self.species):
+                print("Given species does not have that allele.")
```

Comment:
This should probably raise an Exception instead of print and return.

```suggestion
                raise Exception("Given species does not match given allele.")
```